CMakeLists.txt

```diff
@@ -81,9 +84,23 @@ add_definitions (-DBOOST_ASIO_DISABLE_BOOST_DATE_TIME=1)
 add_definitions (-DBOOST_ASIO_DISABLE_BOOST_REGEX=1)
 add_definitions (-DBOOST_COROUTINES_NO_DEPRECATION_WARNING=1)
 
-include_directories (${BOOST_ROOT})
+OPTION(FIND_BOOST "Build beast against an existing Boost install, rather than in-tree")
+if (FIND_BOOST)
+  set(Boost_USE_STATIC_LIBS ON)
```

Comment:
Why are you forcing the use of static libs?

Reply:
If I remember correctly, it was because linking against the dynamic libraries on Windows would then require the Boost DLLs for that version to be in your system path, or be copied into the library directory. The former doesn't appear to be the default, and the latter would normally be handled by `install` rules, and this CMakeLists.txt is not set up for the install stage, and I don't see there's any expectation that this will change.

Basically, this gives the best out-of-the-box experience in common scenarios, where you can immediately run unit-tests (i.e. through CTest, which is a different patch series I haven't put up for merging) after the build. On Windows it'll link to static libs, and (again, if I remember correctly) on Linux you generally don't *have* static libs so it'll link to the dynamic libraries anyway.